### PR TITLE
drivers/sx127x: several fixes

### PR DESCRIPTION
**Pull request by aabadie**
This PR add small fixes to the Semtech SX127x driver:
* setting the modem was incorrect compared to the orignal semtech driver: when uses with their LoRaMAC stack, it was resulting in a crash. Simply applying the same logic fixed this issue
* after transmitting packet (with success or after a timeout), the transceiver was still in 'TX' operating mode. Putting the radio in sleep operating mode in the TX complete/timeout handlers fixes this issue
* other small differences compared with the original driver have been applied in the mean time
* move internal netdev isr callbacks to the right file

Those commits are required for Semtech LoRaMAC integration into RIOT.

### CODE REVIEW

File: drivers/sx127x/sx127x_netdev.c [hunk 1]
```diff
@@ -538,3 +542,155 @@ static int _get_state(sx127x_t *dev, void *val)
...
+            break;
+        case SX127X_MODEM_LORA:
+            /* Clear IRQ */
+            sx127x_reg_write(dev, SX127X_REG_LR_IRQFLAGS,
+                             SX127X_RF_LORA_IRQFLAGS_CADDETECTED |
+                             SX127X_RF_LORA_IRQFLAGS_CADDONE);
+
+            /* Send event message */
+            dev->_internal.is_last_cad_success = (sx127x_reg_read(dev, SX127X_REG_LR_IRQFLAGS) &
+                                                  SX127X_RF_LORA_IRQFLAGS_CADDETECTED) == SX127X_RF_LORA_IRQFLAGS_CADDETECTED;
```
Comment (miri64) [inline drivers/sx127x/sx127x_netdev.c]: Isn't the `== ....` redundant? `SX127X_RF_LORA_IRQFLAGS_CADDETECTED` is just `0x01`. If not, please try to get the line `<100` chars.

**Comment by miri64** [comment]
("tested" now, but only compiled)

File: drivers/sx127x/sx127x_netdev.c (hunk 1 above)
Comment (aabadie) [reply drivers/sx127x/sx127x_netdev.c]: It was already like this before, I just split the lines

**Comment by aabadie** [comment]
@miri64, comments addressed, directly rebased and squashed

File: drivers/sx127x/sx127x_netdev.c [hunk 2]
```diff
@@ -538,3 +542,156 @@ static int _get_state(sx127x_t *dev, void *val)
...
+        case SX127X_MODEM_LORA:
+            /* Clear IRQ */
+            sx127x_reg_write(dev, SX127X_REG_LR_IRQFLAGS,
+                             SX127X_RF_LORA_IRQFLAGS_CADDETECTED |
+                             SX127X_RF_LORA_IRQFLAGS_CADDONE);
+
+            /* Send event message */
+            dev->_internal.is_last_cad_success = ((sx127x_reg_read(dev, SX127X_REG_LR_IRQFLAGS) &
+                                                   SX127X_RF_LORA_IRQFLAGS_CADDETECTED) ==
+                                                  SX127X_RF_LORA_IRQFLAGS_CADDETECTED);
```
Comment (miri64) [inline drivers/sx127x/sx127x_netdev.c]: Still doesn't make sense to check it this way, but ok.

**Pull request merged**